### PR TITLE
Fix bot uid not found

### PR DESCRIPTION
**Pull request by jcfrt**
If the slack "person" is a a bot, a different endpoint has to be used to get information, otherwise the API endpoint "users.info" will return an error saying "user_not_found".

### CODE REVIEW

**Comment by jcfrt** [comment]
This should fix #67

File: _slack/person.py [hunk 1]
```diff
@@ -85,23 +85,37 @@ def _cache_user_info(self):
         if self._userid is None:
             raise ValueError("Unable to look up an undefined user id.")
 
-        res = self._webclient.users_info(user=self._userid)
+        isbot = False
+        if self._userid[0] == "B":
+            isbot = True
+
+        # Fix error while looking up shorter bot uid
+        if len(self._userid) <= 3:
```
Comment (nzlosh) [inline _slack/person.py:93]: Why is this required?  If a bot id is valid, the lookup should work should it?

**Review by nzlosh — COMMENTED** [review]
Thanks for the PR submission.  There's a lot conditional statements which increases the complexity of the code.  Would it be possible to consolidate these conditions to simplify the code?

I think it would be possible to reduce the code to two if conditions without additional variables.  Something like:
```
if self._userid.startswith("B"):
    res = self._webclient.bots_info(bot=self._userid)
else:
    res = self._webclient.users_info(user=self._userid)
...
if "bot" in res:
    self._user_info["display_name"] = res["bot"].get("name", "")
else:
    # existing user code here.
```

File: _slack/person.py (hunk 1 above)
Comment (jcfrt) [reply _slack/person.py:93]: That's a good question. 
I added this check because for some reason I got a lot of errors when running this my Slack instance: requests were made to `slack.com/api/bots.info` with params `{'bot': 'B01'}`, which resulted in a lot of `bot_not_found` errors. 
While this patch fixed my problem, I admit I'm not entirely sure this check should be kept as I do not fully understand what is happening here (these requests seem to be triggered on each command invoked by a user).
I kept it in hoping someone would shed some insight about this.

Comment (jcfrt) [reply _slack/person.py:89]: Actually scratch that, after further inspection these errors were probably not triggered after each user command. 

Here is a sample of the error from the logs:

```
ERROR    slack_sdk.rtm_v2          Failed to run a message listener: The request to the Slack API failed. (url: https://www.slack.com/api/bots.info) args {'headers': {'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'Python/3.10.2 slackclient', 'Authorization': 'Bearer XXXtokenXXX'}, 'data': {}, 'params': {'bot': 'B01'}, 'files': None, 'json': None}
The server responded with: {'ok': False, 'error': 'bot_not_found'}
Traceback (most recent call last):
  File ".venv/lib/python3.10/site-packages/slack_sdk/rtm_v2/__init__.py", line 323, in run_message_listeners
    listener(self, message)
  File "./lib/python3.10/site-packages/slack_sdk/rtm_v2/__init__.py", line 167, in new_message_listener
    func(_self, event)
  File "./backend/err-backend-slackv3/slackv3.py", line 306, in _rtm_generic_event_handler
    return event_handler(client, event)
  File "./backend/err-backend-slackv3/slackv3.py", line 410, in _rtm_handle_message
    self._handle_message(client.web_client, event)
  File "./backend/err-backend-slackv3/slackv3.py", line 537, in _handle_message
    msg.frm = SlackRoomBot(
  File "./backend/err-backend-slackv3/_slack/room.py", line 336, in __init__
    super().__init__(webclient, bot_id, bot_username)
  File "./backend/err-backend-slackv3/_slack/room.py", line 310, in __init__
    super().__init__(webclient, userid=bot_id)
  File "./backend/err-backend-slackv3/_slack/person.py", line 50, in __init__
    self._cache_user_info()
  File "./backend/err-backend-slackv3/_slack/person.py", line 95, in _cache_user_info
    res = self._webclient.bots_info(bot=self._userid)
  File ".venv/lib/python3.10/site-packages/slack_sdk/web/client.py", line 1629, in bots_info
    return self.api_call("bots.info", http_verb="GET", params=kwargs)
  File ".venv/lib/python3.10/site-packages/slack_sdk/web/base_client.py", line 156, in api_call
    return self._sync_send(api_url=api_url, req_args=req_args)
  File ".venv/lib/python3.10/site-packages/slack_sdk/web/base_client.py", line 187, in _sync_send
    return self._urllib_api_call(
  File ".venv/lib/python3.10/site-packages/slack_sdk/web/base_client.py", line 317, in _urllib_api_call
    ).validate()
  File ".venv/lib/python3.10/site-packages/slack_sdk/web/slack_response.py", line 189, in validate
    raise e.SlackApiError(message=msg, response=self)
slack_sdk.errors.SlackApiError: The request to the Slack API failed. (url: https://www.slack.com/api/bots.info) Debug args {'headers': {'Content-Type': 'application/x-www-form-urlencoded', 'User-Agent': 'Python/3.10.2 slackclient', 'Authorization': 'Bearer XXXtokenXXX'}, 'data': {}, 'params': {'bot': 'B01'}, 'files': None, 'json': None}
The server responded with: {'ok': False, 'error': 'bot_not_found'}
```

Comment (jcfrt) [reply _slack/person.py:89]: Sorry (long day), I realize this is the same error message that was originally mentioned in #67. 

Also I had patched `slack_sdk/web/slack_response.py:188` to add the "Debug args" in order to see details about the request being made:

```python
msg = f"The request to the Slack API failed. (url: {self.api_url}) Debug args {self.req_args}"
```

Comment (nzlosh) [reply _slack/person.py:89]: So looking around for "`B01` slack bot" I came across this https://github.com/slackapi/hubot-slack/blob/c0c0d8d348b1522afa9835ef84a46f2bfbc6cdaf/src/client.coffee#L53 ... it seems that `B01` is a special id.  I'll need to find official documentation to understand how this should be correctly handled.

**Comment by nzlosh** [comment]
Thank you for the contribution.  I'll merge with the current logic to skip short bot ids and create a new patch once it's clearer how these magic ids are meant to be handled.
